test/e2e/metering_manual_install_test.go

```diff
@@ -107,6 +110,111 @@ func testManualMeteringInstall(
+		"storageClass": "storageclass.yaml",
+	}
+	server := &corev1.Pod{}
+	service := &corev1.Service{}
+	storageClass := &storagev1beta1.StorageClass{}
+	for name, file := range files {
+		absFile := filepath.Join(repoPath, testNFSManifestPath, file)
+		switch name {
+		case "pod":
+			if err := deploy.DecodeYAMLManifestToObject(absFile, server); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Pods(nfsTestingNamespace).Create(context.TODO(), server, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "service":
+			if err := deploy.DecodeYAMLManifestToObject(absFile, service); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Services(nfsTestingNamespace).Create(context.TODO(), service, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "storageClass":
+			if err := deploy.DecodeYAMLManifestToObject(absFile, storageClass); err != nil {
+				return err
+			}
+			ctx.Client.StorageV1beta1().StorageClasses().Create(context.TODO(), storageClass, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+
+	svc, err := ctx.Client.CoreV1().Services(nfsTestingNamespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return err
+	}
+	if err != nil || svc.Spec.ClusterIP == "" {
+		err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+			svc, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return false, err
+			}
+			if err != nil {
+				return false, nil
+			}
+			return svc.Spec.ClusterIP != "" , nil
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	nfsFile := "persistentvolume.yaml"
+	nfsPv := &corev1.PersistentVolume{}
+	absFile := filepath.Join(repoPath, testNFSManifestPath, nfsFile)
+	//decode pvc
+	if err := deploy.DecodeYAMLManifestToObject(absFile, nfsPv); err != nil {
+		return err
+	}
+
+	nfsPv.Spec.NFS.Server = svc.Spec.ClusterIP
+	_, err = ctx.Client.CoreV1().PersistentVolumes().Create(context.TODO(), nfsPv, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+func createTestingNamespace(client kubernetes.Interface, namespace string) (*corev1.Namespace, error) {
+	ns, err := client.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+	if apierrors.IsNotFound(err) {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+				Labels: map[string]string{
+					"name": namespace + "-" + testingNamespaceLabel,
+				},
+			},
+		}
+		_, err = client.CoreV1().Namespaces().Create(context.Background(), ns, metav1.CreateOptions{})
+		if err != nil {
+			return ns, nil
+		}
+	}
+	return ns, nil
+}
```

Comment:
As a follow-up, we should move this towards a more centralized package as we do something similar in the other tests.

Reply:
Agreed, generalized, this could go in `testhelpers`